src/mca/state/state.h

```diff
@@ -74,12 +80,24 @@ PRTE_EXPORT extern prte_mca_base_framework_t prte_state_base_framework;
         } \
     } while(0);
 
+/* Timestamp for state printouts */
+#define PRTE_STATE_GET_TIMESTAMP(t)                                         \
+    do {                                                                    \
+        struct timeval tv;                                                  \
+        gettimeofday(&tv, NULL);                                            \
+        t  = tv.tv_sec;                                                     \
+        t += (double)tv.tv_usec / 1000000.0;                                \
+    } while(0);
+
 #define PRTE_ACTIVATE_JOB_STATE(j, s)                                       \
     do {                                                                    \
         prte_job_t *shadow=(j);                                             \
+        double timestamp = 0.0;                                             \
```

Comment:
What you do think about adding a wrapper like this around the print block:
```
if( prte_state_base_framework.framework_verbose > 0 ) {
```

Then we don't incur the overhead of the `gettimeofday` on the calls when verbosity is not turned up

Reply:
updated with this change

Reply:
Thanks!